sensor/kubernetes/listener/resources/secrets.go

```diff
@@ -285,26 +298,34 @@ func (s *secretDispatcher) processDockerConfigEvent(secret, oldSecret *v1.Secret
 			continue
 		}
 
+		if skipIntegrationCreate(secret) {
+			log.Debugf("Skipping create of image integration for secret %q, namespace %q, registry %q", secret.GetName(), secret.GetNamespace(), registryAddr)
+			continue
+		}
+
 		ii, err := DockerConfigToImageIntegration(secret, registryAddr, dce)
 		if err != nil {
 			log.Errorf("unable to create docker config for secret %s: %v", secret.GetName(), err)
 			continue
 		}
 
 		sensorEvents = append(sensorEvents, &central.SensorEvent{
-			// Only update is supported at this time.
+			// Only update is supported at this time for integrations without a source.
+			// Set the Id for sourced integrations so that Sensor's deduper (sensor/common/deduper Send())
+			// will NOT suppress subsequent secret remove events.
+			Id:     utils.IfThenElse(sourcedIntegration, ii.GetId(), ""),
```

Comment:
You could also add an exception in the [skipDeduping](https://github.com/stackrox/stackrox/blob/9f3eb2c008a911e3c8b74cb4a2e86ab2259b7421/sensor/common/deduper/deduper.go#L113) function.

Reply:
Oh Interesting, I didn't see that. 

I'm thinking it may still be advantageous to keep the logic where it is, so that we get the benefit of deduping (assuming the deduper is valuable) and the overhead of checking the feature, env var, and integration source is limited to only the image integration events vs. adding a little bit of overhead to every execution of the deduper (which admittedly may be negligible).

Reply:
That makes sense. No objections to keeping it here 🙂